### ISSUE TITLE
Removing excess Compathelper branches

### ISSUE THREAD

**Issue by aminya** [JuliaRegistries/CompatHelper.jl#181]
I think CompatHelper should remove the branches once their PR is merged.
Similar to how https://github.com/marketplace/actions/create-pull-request does it.

**Comment by DilumAluthge** [comment]
Why not just use GitHub's built-in feature: https://help.github.com/en/github/administering-a-repository/managing-the-automatic-deletion-of-branches

**Comment by aminya** [comment]
Because I only want to do this for CompatHelper branches.

**Comment by DilumAluthge** [comment]
The tricky part is that in order for this to work, CompatHelper would need to be triggered every time you merge aPR. So every time that you merge a PR, CompatHelper would need to run, check and see if you merged a CompatHelper PR, and then, if it was a CompatHelper PR that was merged, then it would delete the branch.

I'm totally fine with adding this functionality, but it will need someone to code it up and make a PR to this repo. Unfortunately, I don't have the time to do it myself, but I can review a PR if someone else makes it.

**Comment by DilumAluthge** [comment]
> Note: I think that we need to move the part of the code that creates Pull request to Github.jl so everyone can use it.

Sure! The CompatHelper.jl code is MIT-licensed, so anyone is welcome to take the code from this repo and add it to the GitHub.jl repository.

**Comment by DilumAluthge** [comment]
I'm going to close this as `wontfix` because of this part:
> because this functionality would mean a lot more CompatHelper runs (it would run every time anyone merges any PR)
